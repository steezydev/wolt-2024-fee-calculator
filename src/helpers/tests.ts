const localStorageMock = (function () {
    let store: Record<string, string> = {};
    return {
        getItem(key: string) {
            return store[key] || null;
        },
        setItem(key: string, value: string) {
            store[key] = value.toString();
        },
        clear() {
            store = {};
        },
    };
})();

const matchMediaMock = jest.fn().mockImplementation((query) => ({
    matches: query === '(prefers-color-scheme: dark)',
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
}));

export const mockContext = () => {
    beforeAll(() => {
        Object.defineProperty(window, 'localStorage', {
            value: {
                getItem: jest.fn().mockImplementation((key: string) => {
                    return localStorageMock.getItem(key);
                }),
                setItem: jest
                    .fn()
                    .mockImplementation((key: string, value: string) => {
                        localStorageMock.setItem(key, value);
                    }),
                clear: localStorageMock.clear,
            },
            writable: true,
        });

        window.matchMedia = matchMediaMock;
    });

    beforeEach(() => {
        (window.localStorage.getItem as jest.Mock).mockClear();
        (window.localStorage.setItem as jest.Mock).mockClear();
        matchMediaMock.mockClear();
    });
};

export const setupFakeDate = () => {
    beforeEach(() => {
        jest.useFakeTimers();
        jest.setSystemTime(new Date('2024-01-01'));
    });

    afterEach(() => {
        jest.useRealTimers();
    });
};

export const mockScrollTo = () => {
    beforeEach(() => {
        Element.prototype.scrollTo = jest.fn();
    });

    afterEach(() => {
        (Element.prototype.scrollTo as jest.Mock).mockRestore();
    });
};
