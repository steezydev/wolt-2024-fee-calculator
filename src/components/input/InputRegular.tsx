import InputUnstyled from '@/components/input/InputUnstyled';
import { classNames } from '@/helpers/classNames';
import { InputProps } from '@/types/props/InputProps';
import React from 'react';

const InputRegular = React.forwardRef<HTMLInputElement, InputProps>(
  ({ className, ...props }, ref) => {
    return (
      <InputUnstyled
        ref={ref}
        {...props}
        className={classNames(
          'rounded-xl border-2 bg-transparent border-neutral-light-100 placeholder:text-neutral-light-300 px-3 py-2.5 text-xl text-black leading-none',
          ' dark:border-neutral-dark-100 dark:placeholder:text-neutral-dark-300 dark:text-white',
          'group-hover:data-[invalid=false]:border-primary-300',
          'data-[invalid=true]:placeholder:text-error data-[invalid=true]:dark:placeholder:text-error',
          'data-[invalid=true]:border-error data-[invalid=true]:dark:border-error focus:data-[invalid=true]:border-primary-300 focus:data-[invalid=true]:dark:border-primary-300',
          'focus:outline-none focus:ring-1 focus:ring-primary-300 focus:border-primary-300 focus:dark:border-primary-300',
          className
        )}
      />
    );
  }
);

// const InputRegular = ({ className, ...props }: InputProps) => {
//   return (
//     <InputUnstyled
//       {...props}
//       className={classNames(
//         'rounded-xl border-2 bg-transparent border-neutral-light-100 placeholder:text-neutral-light-300 px-3 py-2.5 text-xl text-black leading-none',
//         ' dark:border-neutral-dark-100 dark:placeholder:text-neutral-dark-300 dark:text-white',
//         'group-hover:data-[invalid=false]:border-primary-300',
//         'data-[invalid=true]:placeholder:text-error data-[invalid=true]:dark:placeholder:text-error',
//         'data-[invalid=true]:border-error data-[invalid=true]:dark:border-error focus:data-[invalid=true]:border-primary-300 focus:data-[invalid=true]:dark:border-primary-300',
//         'focus:outline-none focus:ring-1 focus:ring-primary-300 focus:border-primary-300 focus:dark:border-primary-300',
//         className
//       )}
//     />
//   );
// };

export default InputRegular;
