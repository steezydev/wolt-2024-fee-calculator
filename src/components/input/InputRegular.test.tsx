import { inputRenderTests } from './Input.tests';
import InputRegular from './InputRegular';

inputRenderTests('InputRegular', InputRegular);
