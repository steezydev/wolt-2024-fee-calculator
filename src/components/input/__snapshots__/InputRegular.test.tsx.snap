// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputRegular when asked to render autofocused renders autofocused 1`] = `
<input
  aria-invalid={false}
  aria-label="Test label"
  aria-required={false}
  autoFocus={true}
  className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none rounded-xl border-2 bg-transparent border-neutral-light-100 placeholder:text-neutral-light-300 px-3 py-2.5 text-xl text-black leading-none  dark:border-neutral-dark-100 dark:placeholder:text-neutral-dark-300 dark:text-white group-hover:data-[invalid=false]:border-primary-300 data-[invalid=true]:border-error data-[invalid=true]:dark:border-error focus:data-[invalid=true]:border-primary-300 focus:data-[invalid=true]:dark:border-primary-300 focus:outline-none focus:ring-1 focus:ring-primary-300 focus:border-primary-300 focus:dark:border-primary-300"
  data-invalid={false}
  data-test-id="test-input"
  disabled={false}
  id="test-input"
  name="test-input"
  required={false}
  type="text"
/>
`;

exports[`InputRegular when asked to render disabled renders disabled 1`] = `
<input
  aria-invalid={false}
  aria-label="Test label"
  aria-required={false}
  autoFocus={false}
  className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none rounded-xl border-2 bg-transparent border-neutral-light-100 placeholder:text-neutral-light-300 px-3 py-2.5 text-xl text-black leading-none  dark:border-neutral-dark-100 dark:placeholder:text-neutral-dark-300 dark:text-white group-hover:data-[invalid=false]:border-primary-300 data-[invalid=true]:border-error data-[invalid=true]:dark:border-error focus:data-[invalid=true]:border-primary-300 focus:data-[invalid=true]:dark:border-primary-300 focus:outline-none focus:ring-1 focus:ring-primary-300 focus:border-primary-300 focus:dark:border-primary-300"
  data-invalid={false}
  data-test-id="test-input"
  disabled={true}
  id="test-input"
  name="test-input"
  required={false}
  type="text"
/>
`;

exports[`InputRegular when asked to render renders 1`] = `
<input
  aria-invalid={false}
  aria-label="Test label"
  aria-required={false}
  autoFocus={false}
  className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none rounded-xl border-2 bg-transparent border-neutral-light-100 placeholder:text-neutral-light-300 px-3 py-2.5 text-xl text-black leading-none  dark:border-neutral-dark-100 dark:placeholder:text-neutral-dark-300 dark:text-white group-hover:data-[invalid=false]:border-primary-300 data-[invalid=true]:border-error data-[invalid=true]:dark:border-error focus:data-[invalid=true]:border-primary-300 focus:data-[invalid=true]:dark:border-primary-300 focus:outline-none focus:ring-1 focus:ring-primary-300 focus:border-primary-300 focus:dark:border-primary-300"
  data-invalid={false}
  data-test-id="test-input"
  disabled={false}
  id="test-input"
  name="test-input"
  required={false}
  type="text"
/>
`;

exports[`InputRegular when asked to render required renders required 1`] = `
<input
  aria-invalid={false}
  aria-label="Test label"
  aria-required={true}
  autoFocus={false}
  className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none rounded-xl border-2 bg-transparent border-neutral-light-100 placeholder:text-neutral-light-300 px-3 py-2.5 text-xl text-black leading-none  dark:border-neutral-dark-100 dark:placeholder:text-neutral-dark-300 dark:text-white group-hover:data-[invalid=false]:border-primary-300 data-[invalid=true]:border-error data-[invalid=true]:dark:border-error focus:data-[invalid=true]:border-primary-300 focus:data-[invalid=true]:dark:border-primary-300 focus:outline-none focus:ring-1 focus:ring-primary-300 focus:border-primary-300 focus:dark:border-primary-300"
  data-invalid={false}
  data-test-id="test-input"
  disabled={false}
  id="test-input"
  name="test-input"
  required={true}
  type="text"
/>
`;

exports[`InputRegular when asked to render with placeholder renders with placeholder 1`] = `
<input
  aria-invalid={false}
  aria-label="Test label"
  aria-required={false}
  autoFocus={false}
  className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none rounded-xl border-2 bg-transparent border-neutral-light-100 placeholder:text-neutral-light-300 px-3 py-2.5 text-xl text-black leading-none  dark:border-neutral-dark-100 dark:placeholder:text-neutral-dark-300 dark:text-white group-hover:data-[invalid=false]:border-primary-300 data-[invalid=true]:border-error data-[invalid=true]:dark:border-error focus:data-[invalid=true]:border-primary-300 focus:data-[invalid=true]:dark:border-primary-300 focus:outline-none focus:ring-1 focus:ring-primary-300 focus:border-primary-300 focus:dark:border-primary-300"
  data-invalid={false}
  data-test-id="test-input"
  disabled={false}
  id="test-input"
  name="test-input"
  placeholder="Test placeholder"
  required={false}
  type="text"
/>
`;

exports[`InputRegular when asked to render with value renders with placeholder 1`] = `
<input
  aria-invalid={false}
  aria-label="Test label"
  aria-required={false}
  autoFocus={false}
  className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none rounded-xl border-2 bg-transparent border-neutral-light-100 placeholder:text-neutral-light-300 px-3 py-2.5 text-xl text-black leading-none  dark:border-neutral-dark-100 dark:placeholder:text-neutral-dark-300 dark:text-white group-hover:data-[invalid=false]:border-primary-300 data-[invalid=true]:border-error data-[invalid=true]:dark:border-error focus:data-[invalid=true]:border-primary-300 focus:data-[invalid=true]:dark:border-primary-300 focus:outline-none focus:ring-1 focus:ring-primary-300 focus:border-primary-300 focus:dark:border-primary-300"
  data-invalid={false}
  data-test-id="test-input"
  disabled={false}
  id="test-input"
  name="test-input"
  required={false}
  type="text"
  value="Test value"
/>
`;
